sql/plan/join.go

```diff
@@ -176,6 +180,18 @@ func (i JoinType) IsCross() bool {
 	return i == JoinTypeCross || i == JoinTypeCrossHash
 }
 
+func (i JoinType) IsLateralCrossJoin() bool {
```

Comment:
Do we need more than `IsLateral?`? Do `IsCross` and `IsLeft` need the new types?